lib/ecto/repo/preloader.ex

```diff
@@ -104,8 +104,8 @@ defmodule Ecto.Repo.Preloader do
       # set the proper timeouts.
       opts = Keyword.put_new(opts, :caller, self())
       assocs
-      |> Enum.map(&Task.async(:erlang, :apply, [fun, [&1, opts]]))
-      |> Enum.map(&Task.await(&1, :infinity))
+      |> Task.async_stream(&fun.(&1, opts))
+      |> Enum.map(fn {:ok, assoc} -> assoc end)
```

Comment:
Let's use `Stream.map` - we have a stream anyway.

Reply:
@michalmuskala thanks, changed.

Reply:
I think it's better evaluate the function in one place instead of somtimes returning a list and sometimes returning a stream. Enum.map should be faster as well if that matters.

Reply:
@michalmuskala we should stick with `Enum.map` because we want to realize the computation now and not later on.